parley-todo.md

```diff
@@ -0,0 +1,148 @@
+        - I think this... just works already?
+    - [x] finish the gnarly parts that i've been putting off
+      - [x] indentation (done, but untested because parley's tab character support is broken)
+      - [x] selecting a range without having a Galley rendered already (done)
+    - [x] Fully remove CCursor and CCursorRange
+  - [x] Support `char_limit` (done but untested)
+  - [x] AccessKit integration(?) (done; kinda janky and cannot currently test whether the bounding boxes are correct)
+  - [x] Remove RowVertexIndices from selection painting
+  - [ ] Do another pass over TextBuffer's API
+  - [ ] Test IME support
+  - [ ] Smoothe out AccessKit API integration (and reduce temp allocations)
+  - [ ] Test AccessKit text bounding boxes (horiz_offset for alignment and vertical_offset for wrapped labels working)
+  - [ ] Character-based cursors could never be in an invalid state; byte-based cursors can. Ensure we don't try to slice a string using any non-validated selections
+- [ ] Text styling
+  - [x] Fix FontDefinitions and adding fonts
+    - [x] Get fallback/ordering working properly
+      - I think this is done?
+  - [x] Auto fallback to faux italics (and perhaps faux bold)
+    - [x] run.synthesis()
+    - Faux bold doesn't work because Parley always tries to embolden Ubuntu Light because we ask for Ubuntu Regular
+  - [ ] Actually render text decorations (underline, strikethrough, etc) and backgrounds
+    - [x] Strikethrough and underline
+      - [x] Feathered
+    - [x] Backgrounds
+    - [x] valign ~~(this will take a lot of implementing in Parley)~~
+    - [x] Families and not just files
+    - [x] Option to load system fonts in FontDefinitions
+  - [ ] FontTweak is very not-implemented
+    - [ ] scale
+      - This actually *does* affect layout, but only horizontally. Not sure if this is implementable.
+    - [x] y_offset_factor
+    - [x] y_offset
+    - [ ] baseline_offset_factor
+      - What does this even do?
+  - [x] Can we do bold now?
+  - [x] Letter spacing
+    - Why was this here? It's been implemented for a while
+  - [x] Hinting enable/disable
+    - [x] Maybe a global setting and also an override in FontTweak?
+  - [ ] If file size isn't an issue, ship variable fonts in epaint-default-fonts
+  - [x] Make sure to test with syntect disabled (there's some old style-heavy code that is cfg'd out with syntect enabled)
+  - [ ] Smoothe out the janky parts of the new API
+    - [ ] For mixed-DPI purposes, and because we don't need to store the FontStore as a mutex, ctx.fonts() now returns a "fonts view" that's technically read/write. But there are no operations that *semantically* modify the fonts from it
+    - [ ] FontStore and Fonts are different and we should just expose them separately instead of passing through all the FontStore methods onto Fonts
+  - [ ] Work around https://github.com/jslegers/emoji-icon-font/issues/18 / https://github.com/emilk/egui/issues/1284
```

Comment:
I opened https://github.com/jslegers/emoji-icon-font/pull/19 to fix this and it was merged.

Reply:
So I assume https://github.com/emilk/egui/blob/d78fc39386e28ec7a18bd48a28a688177f090bd4/crates/epaint_default_fonts/fonts/emoji-icon-font.ttf needs to be replaced with the new ttf file from https://github.com/jslegers/emoji-icon-font/blob/master/fonts/emoji.ttf

Reply:
https://github.com/emilk/egui/pull/5877

Reply:
emilk fixed this in https://github.com/emilk/egui/pull/7067